daemon/cmd/local_node_sync.go

```diff
@@ -54,6 +55,11 @@ func (ini *localNodeSynchronizer) InitLocalNode(ctx context.Context, n *node.Loc
 	if err := ini.initFromK8s(ctx, n); err != nil {
 		return err
 	}
+
+	if ini.WireGuard != nil {
```

Comment:
This check should not be necessary, as it is not an optional dependency.

Reply:
So if WireGuard is disabled, we return a nil-valued point here:
https://github.com/cilium/cilium/blob/835e9613d43a4ebbeb6a1dfafcfd4238cbe29bc8/pkg/datapath/cells.go#L155

I guess there is a difference between a `nil` value being provided (which can be done for required values) and no value being provided (which only works for optional values).

Reply:
Right, I missed it could legitimately be nil.